Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/factory.go

```diff
@@ -209,6 +209,9 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				}
 				return &clientSwaggerSchema{client, api.Scheme}, nil
 			}
+			if filenameFlag := flags.Lookup("filename"); filenameFlag != nil && filenameFlag.Changed {
```

Comment:
ToJSON(data, checkValid) makes sense to me.  I don't know that the validator is *wrong*, so it's fine for now.  We can come back and fix it later - please open an upstream issue so that we don't forget to fix it.

----- Original Message -----
> > @@ -209,6 +209,9 @@ func NewFactory(optionalClientConfig
> > clientcmd.ClientConfig) *Factory {
> >  				}
> >  				return &clientSwaggerSchema{client, api.Scheme}, nil
> >  			}
> > +			if filenameFlag := flags.Lookup("filename"); filenameFlag != nil &&
> > filenameFlag.Changed {
> 
> > This forces filename validation on all objects processed
> 
> Yeah, that's the caveat of doing it this way. Today `create`, `update`,
> `rollingupdate`, etc all use the resource builder just once. But this could
> introduce an issue in future.
> 
> Any suggestion about another way to use the validator, or should we just
> switch the approach, e.g. in `ToJSON`?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2610/files#r31646531